src/Features/Core/Portable/Completion/Providers/AbstractTypeImportCompletionProvider.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.AddImports;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Experiments;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal abstract partial class AbstractTypeImportCompletionProvider : CommonCompletionProvider
+    {
+        protected abstract Task<SyntaxContext> CreateContextAsync(Document document, int position, CancellationToken cancellationToken);
+
+        protected abstract ImmutableArray<string> GetImportedNamespaces(
+            SyntaxNode location,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken);
+
+        public override async Task ProvideCompletionsAsync(CompletionContext completionContext)
+        {
+            var cancellationToken = completionContext.CancellationToken;
+            var document = completionContext.Document;
+            var workspace = document.Project.Solution.Workspace;
+            var experimentationService = workspace.Services.GetService<IExperimentationService>();
+
+            var importCompletionOptionValue = completionContext.Options.GetOption(CompletionOptions.ShowImportCompletionItems, document.Project.Language);
+
+            // Don't trigger import completion if the option value is "default" and the experiment is disabled for the user. 
+            if (importCompletionOptionValue == false ||
+                (importCompletionOptionValue == null && experimentationService?.IsExperimentEnabled(WellKnownExperimentNames.TypeImportCompletion) != true))
+            {
+                return;
+            }
+
+            var syntaxContext = await CreateContextAsync(document, completionContext.Position, cancellationToken).ConfigureAwait(false);
+            if (!syntaxContext.IsTypeContext)
+            {
+                return;
+            }
+
+            using (Logger.LogBlock(FunctionId.Completion_TypeImportCompletionProvider_GetCompletionItemsAsync, cancellationToken))
+            {
+                await AddCompletionItemsAsync(completionContext, syntaxContext, cancellationToken).ConfigureAwait(false);
+            }
+        }
+
+        private async Task AddCompletionItemsAsync(CompletionContext completionContext, SyntaxContext syntaxContext, CancellationToken cancellation)
```

Comment:
you have 'cancellation' and 'cancellationToken'

Reply:
fixed